beacon_node/network/src/attestation_service/mod.rs

```diff
@@ -358,6 +326,47 @@ impl<T: BeaconChainTypes> AttestationService<T> {
         Ok(())
     }
 
+    /// Checks if we have a discover peers message queued and sends a message if necessary
+    ///
+    /// If a message exists for the same subnet, compare the `min_ttl` of the current and
+    /// queued messages and extend the queued message.
+    fn queue_discover_peers_message(&mut self, subnet_id: SubnetId, min_ttl: Option<Instant>) {
+        let message = AttServiceMessage::DiscoverPeers { subnet_id, min_ttl };
+        self.events.iter_mut().for_each(|event| {
+            match event {
+                AttServiceMessage::DiscoverPeers {
+                    subnet_id: other_subnet_id,
+                    min_ttl: other_min_ttl,
+                } => {
+                    if subnet_id == *other_subnet_id {
+                        let other_min_ttl_clone = other_min_ttl.clone();
+                        match (min_ttl, other_min_ttl_clone) {
+                            (Some(min_ttl_instant), Some(other_min_ttl_instant)) =>
+                            // only update the min_ttl if it is greater than the existing min_ttl and a DURATION_DIFFERENCE padding
+                            {
+                                if min_ttl_instant.saturating_duration_since(other_min_ttl_instant)
+                                    > DURATION_DIFFERENCE
+                                {
+                                    *other_min_ttl = min_ttl;
+                                    return;
+                                }
+                            }
+                            (None, Some(_)) => {
+                                // Update the min_ttl to None, because the new message is longer-lived.
+                                *other_min_ttl = None;
+                                return;
+                            }
+                            (Some(_), None) => {} // Don't replace this because the existing message is for a longer-lived peer.
+                            (None, None) => {}    // Duplicate message.
+                        }
+                    }
+                }
+                _ => {}
+            };
+        });
+        self.events.push_back(message);
```

Comment:
Even if this is a duplicate, or if we've just updated an event already in the queue, won't this still add this message to the queue, effectively duplicating?